core/src/state.rs

```diff
@@ -899,7 +865,34 @@ impl WorldTransaction<'_, '_> {
 
     /// Set executor data model.
     pub fn set_executor_data_model(&mut self, executor_data_model: ExecutorDataModel) {
-        *self.executor_data_model.get_mut() = executor_data_model;
+        let prev_executor_data_model =
+            core::mem::replace(self.executor_data_model.get_mut(), executor_data_model);
+
+        self.update_parameters(&prev_executor_data_model);
```

Comment:
unfortunately, unlike for permissions, here I have to do custom migration on the core side. There is no ISI for removing parameters. Therefore, I both remove parameters and set new ones to default values